src/view.c

```diff
@@ -874,6 +875,69 @@ view_center(struct view *view, const struct wlr_box *ref)
 	}
 }
 
+/*
+ * Algorithm copied from KWin
+ * https://github.com/KDE/kwin/blob/df9f8f8346b5b7645578e37365dabb1a7b02ca5a/src/placement.cpp#L589
+ */
+static void
+view_cascade(struct view *view)
+{
+	/* Default geometry to which the view is moved */
+	struct wlr_box center = view->pending;
+	view_compute_centered_position(view, NULL,
+		center.width, center.height, &center.x, &center.y);
+
+	/* SSD-aware candidate geometry to which the view is moved */
+	struct wlr_box self = center;
+	struct border margin = ssd_get_margin(view->ssd);
+	self.x -= margin.left;
+	self.y -= margin.top;
+	self.width += margin.left + margin.right;
+	self.height += margin.top + margin.bottom;
+
+	struct wlr_box usable = output_usable_area_in_layout_coords(view->output);
+
+cascade_if_covers:;
+	struct wlr_box covered = {0};
+
+	/* Iterate over views from top to bottom */
+	struct view *other_view;
+	for_each_view(other_view, &view->server->views,
+			LAB_VIEW_CRITERIA_CURRENT_WORKSPACE) {
+		struct wlr_box other = ssd_max_extents(other_view);
+		if (other_view == view || !wlr_box_intersects(&self, &other)) {
+			continue;
+		}
```

Comment:
Does this miss a `other_view->minimized` check?

Reply:
You're right. Will fix.

Reply:
Done.